stats/lib/app.js

```diff
@@ -0,0 +1,151 @@
+import * as Sentry from '@sentry/node'
+import Fastify from 'fastify'
+import cors from '@fastify/cors'
+import urlData from '@fastify/url-data'
+
+import { withFilter } from './request-helpers.js'
+
+import {
+  fetchDailyDealStats,
+  fetchDailyParticipants,
+  fetchMinersRSRSummary,
+  fetchMonthlyParticipants,
+  fetchParticipantChangeRates,
+  fetchParticipantScheduledRewards,
+  fetchParticipantRewardTransfers,
+  fetchRetrievalSuccessRate,
+  fetchDealSummary,
+  fetchDailyRetrievalResultCodes,
+  fetchDailyMinerRSRSummary,
+  fetchDailyRetrievalTimings,
+  fetchDailyMinerRetrievalTimings,
+  fetchMinersTimingsSummary
+} from './stats-fetchers.js'
+
+import { addPlatformRoutes } from './platform-routes.js'
+
+/** @typedef {import('@filecoin-station/spark-stats-db').PgPools} PgPools */
+/** @typedef {import('./typings.js').DateRangeFilter} DateRangeFilter */
+/** @typedef {import('./typings.js').RequestWithFilter} RequestWithFilter */
+/** @typedef {import('./typings.js').RequestWithFilterAndAddress} RequestWithFilterAndAddress */
+/** @typedef {import('./typings.js').RequestWithFilterAndAddress} RequestWithFilterAndMinerId */
+
+/**
+ * @param {object} args
+ * @param {string} args.SPARK_API_BASE_URL
+ * @param {import('@filecoin-station/spark-stats-db').PgPools} args.pgPools
+ * @param {boolean} args.logger
+ * @returns
+ */
+export const createApp = ({
+  SPARK_API_BASE_URL,
+  pgPools,
+  logger
+}) => {
+  const app = Fastify({ logger })
+  Sentry.setupFastifyErrorHandler(app)
+
+  app.register(cors, {
+    origin: ['http://localhost:3000', 'app://-']
+  })
+  app.register(urlData)
+
+  app.get('/deals/daily', async (/** @type {RequestWithFilter} */ request, reply) => {
+    await withFilter(request, reply, filter => {
+      return fetchDailyDealStats(pgPools, filter)
+    })
+  })
+  app.get('/deals/summary', async (/** @type {RequestWithFilter} */ request, reply) => {
```

Comment:
I feel like we could maybe introduce new helper for registering a handler as most of the handlers use `withFilter` helper anyways.

Reply:
Did that in https://github.com/filecoin-station/spark-stats/pull/293/commits/abff832be3ab4c4d8e1326ae48276870b540aec4 :)

Reply:
Nice, that looks much cleaner than the thing we had before, good job!